service/controller/clusterapi/v17/resources/machinedeploymentstatus/create.go

```diff
@@ -2,8 +2,103 @@ package machinedeploymentstatus
 
 import (
 	"context"
+
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	"github.com/giantswarm/tenantcluster"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/controllercontext"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToMachineDeployment(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var nodes []corev1.Node
+	var ready []corev1.Node
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding nodes of tenant cluster")
+
+		l, err := cc.Client.TenantCluster.K8s.CoreV1().Nodes().List(metav1.ListOptions{})
+		if tenantcluster.IsTimeout(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "timeout fetching certificates")
+
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
While implementing the status for cluster, I was thinking if there's any use to execute rest of the resources on case of cancellation or if it makes more sense to actually completely skip the current reconciliation cycle?

Reply:
Depends on the circumstances and reasons you may have in mind. I would follow the approach we have so far. I was already thinking to have dedicated controllers for only executing status updating resources but thinking ahead using the level based API approach like in `kubebuilder` this would may not make any difference or sense at all.

Reply:
My thinking here was that if tenant cluster client fails to timeout, probability to hit this again later in other resources is pretty high and that only delays other reconciliation unnecessarily so for me it would make sense to give up and wait for refresh.

Reply:
When other resources would deal with the tenant API then yes. I do not see which resource should do that. Sounds made up and I prefer consistency over premature optimization. Let's iterate here when there is a real reason.

Reply:
`resourcecanceledcontext. SetCanceled ` call  is not needed for basic resource.

Reply:
Good catch buddy. Removed.